distributed/protocol/core.py

```diff
@@ -15,6 +21,39 @@
 
 logger = logging.getLogger(__name__)
 
+cache_dumps = LRU(maxsize=100)
+cache_loads = LRU(maxsize=100)
+cache_dumps_lock = threading.Lock()
+cache_loads_lock = threading.Lock()
+
+
+def dumps_function(func):
+    """ Dump a function to bytes, cache functions """
+    try:
+        with cache_dumps_lock:
+            result = cache_dumps[func]
+    except KeyError:
+        result = pickle.dumps(func, protocol=4)
+        if len(result) < 100000:
```

Comment:
I think we can be a bit more generous with the cache size. currently we're at `100 (LRU maxsize) * 100_000 B (result) ~ 1MB`. Considering how much stuff we're logging without taking size into account too much, I would suggest to be more generous with this upper limit since large results are the juicy cache hits.

Reply:
It sounds like that was just copied and moved over from here

https://github.com/dask/distributed/blob/fa5d9934974fa2a1a3c4b470b6ed2a6b3d4b66d5/distributed/worker.py#L3528

Perhaps we can make a new issue and revisit?

Reply:
> Perhaps we can make a new issue and revisit?

My plan is to remove `worker.dumps_function()` completely, it shouldn't be required to call it explicitly.

Reply:
Ah ok in that case I don't think the `protocol=4` bit above will be needed